index.js

```diff
@@ -0,0 +1,176 @@
+}, {});
+
+//5. Создать массив с продуктами, которые доступны.
+//prettier-ignore
+const filterByAvailable = products.filter(({available})=>available)
+
+//6. Создать функцию, которая принимает массив продуктов и строку = название ингредиента, и возвращает массив с продуктами, где содержится такой ингредиент.
+//prettier-ignore
+const findProductByIngredient = (enterArray, substring) => products.filter((product) => product.ingredients.includes(substring));
+
+//7. Создать функцию, которая принимает массив продуктов и массив ингредиентов, и возвращает массив с продуктами, где содержатся такие ингредиенты.
+const findProductByIngredients = (enterArray, enterIngredients) => {
+  return enterArray.reduce((acc, product) => {
+    const { ingredients } = product;
+    return ingredients.some((el) => enterIngredients.indexOf(el) >= 0)
+      ? [...acc, product]
+      : [...acc];
+  }, []);
+};
+
+//8. Создать функцию, которая принимает массив продуктов и цену, и возвращает массив продуктов, где цена продукта ниже или равна цене из второго аргумента функции.
+const filterProductsByPrice = (enterArray, price) => {
+  return enterArray.filter((product) => product.price <= price);
+};
+
+//9. Создать функцию, которая принимает массив продуктов и массив айдишников, и возвращает строку, где строка включает в себя название продуктов и их цену через запятую, у которых айдишники совпадают.
+const stringFromProducts = (enterArray, enterId) => {
+  return enterArray.reduce((acc, product) => {
+    const { id, name, price, currency } = product;
+    return enterId.includes(id)
+      ? (acc = acc + `${name}:${price} ${currency}, `)
+      : acc;
+  }, "");
+};
+
+//10. Создать функцию, которая принимает массив продуктов и массив айдишников, и возвращает объект, c общими суммами цен продуктов(у которых айдишники совпадают) по каждой валюте.
+const objectFromProductsByCurrency = (enterArray, enterId) => {
+  return enterArray.reduce(
+    (acc, product) => {
+      const { id, price, currency } = product;
+      if (enterId.includes(id)) {
+        currency === "usd"
+          ? (acc = { ...acc, usd: acc.usd + price })
+          : (acc = { ...acc, euro: acc.euro + price });
+      }
+      return acc;
+    },
+    { euro: 0, usd: 0 }
+  );
+};
+
+//11. Создать функцию, которая принимает массив продуктов и массив айдишников, и return строку, где число равно сумме цен продуктов + значок валюты. При этом если, у нас попадают продукты с разными валютами, то мы должны получить сумму в евро и перевести доллары в евро(использовать для этого курс евро/доллар).
+const getSumOfAllProductsById = (enterArray, enterId) => {
+  const summa = enterArray.reduce(
+    (acc = { currency, total }, product) => {
+      const { id, price, currency } = product;
+      if (enterId.includes(id)) {
+        acc.currency !== currency ? (acc.currency = "euro") : null;
+        return currency === acc.currency
+          ? (acc = { ...acc, total: acc.total + price })
```

Comment:
again mutation (acc = { ...acc, total: acc.total + price })
acc.currency !== currency ? (acc.currency = "euro") : null;
: (acc = { ...acc, total: acc.total + price * 0.85 });
we only return **acc**